### DISCUSSION TITLE
Team, should we be concerned with logs presenting themselves interleaved or out-of-sequence?

### DISCUSSION THREAD

**Discussion by deostroll** [nodejs/help#4663] [Q&A]
Is it a sign of unhealthiness in an application?

There is an app that uses a database access framework we built. That module has logging calls. We use the node-buynan logger, at times there are a few console.log() calls too. There is a specific set of sub tasks which should produce logs in a defined sequence - say, 1, 2, 3, 4, 5. These tasks are serial - they happen one-after-another. But in a particular app which consumes our framework the logs from the same module appear in a different order (and, it is random). Sometimes, 5, 3, 4, 2, 1; or sometimes 3, 4, 5, 1, 2. Sometimes the logs are interleaved. Meaning, somewhere middle of a long log line, there are logs coming from a different module, and, then the previous log lines resume.

How to understand why in this particular app the logs present themselves this way? Is this worth chasing?

**Comment by avivkeller** [comment]
Hi! These discussions are specific to the Node.js core, for general code questions, please use other forums, such as StackExchange.

Thank you!
